xdsl/parser/attribute_parser.py

```diff
@@ -1124,6 +1125,15 @@ def try_parse_builtin_boolean_attr(
             return IntegerAttr(1 if value else 0, IntegerType(1))
         return None
 
+    def try_parse_builtin_none_type(self) -> NoneType | None:
+        """
+        Parse a none type, if present
+        none-type ::= `none`
+        """
+        if (value := self.parse_optional_type()) is not None:
+            return NoneType(value)
```

Comment:
I would recommend looking into how `IndexType` is parsed, and adding similar logic there. This won't work because parse_optional_type should itself be able to pase `none` and return `NoneType`